### PR TITLE
Feature/932 noxgenerator noxui update noxgenerator to allow saving files with systemiofile

### LINKED ISSUE #932: [Nox.Ui] Add Nox to Cryptocash.UI
1. Move all prepared for generation files to a new folder (ie Generated.Prepared)
2. Add Nox to the Cryprocash.UI
3. Make sure .razor files are generated and can be used it other views.

FYI @rochar @DanielHodgeGH

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

File: src/Nox.Generator/Nox.Generator.cs [hunk 1]
```diff
@@ -205,4 +207,11 @@ private void GenerateSource(SourceProductionContext context, ImmutableArray<(str
 
         return true;
     }
+
+    private static string? GetProjectRootDirectory(ImmutableArray<(string Path, SourceText? Source)> noxYamls)
+    {
+        var generatorName = "\\generator.nox.yaml";
+        var generatorPath = noxYamls.FirstOrDefault(x => x.Path.EndsWith(generatorName)).Path;
+        return generatorPath?.Remove(generatorPath.Length - generatorName.Length);
```
Comment (rochar) [inline src/Nox.Generator/Nox.Generator.cs]: see the Path utility and for example https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getpathroot?view=net-7.0

File: src/Nox.Generator/Nox.Generator.cs [hunk 2]
```diff
@@ -75,11 +75,13 @@ private void GenerateSource(SourceProductionContext context, ImmutableArray<(str
                     .Select(x => (INoxCodeGenerator)Activator.CreateInstance(x))
                     .ToArray();
 
+                var projectRoot = GetProjectRootDirectory(noxYamls) ?? "";
```
Comment (rochar) [inline src/Nox.Generator/Nox.Generator.cs]: make this nullable, let's be explicitly to be clear how to use it

File: src/Nox.Generator/Application/Commands/CreateCommandGenerator.cs
Comment (rochar) [inline src/Nox.Generator/Application/Commands/CreateCommandGenerator.cs:1]: I would suggest that you rename outputPath to projectRootPath to be explicitly what is being inputed

**Review by rochar — CHANGES_REQUESTED** [review]

**Comment by anna-naumova25** [comment]
@rochar 
I've updated the PR based on all the comments.

**Review by rochar — APPROVED** [review]
good job

**Review by ILoveMyTortoise — APPROVED** [review]

**Pull request merged**